strawberryfields/api_client.py

```diff
@@ -0,0 +1,544 @@
+# Copyright 2019 Xanadu Quantum Technologies Inc.
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+#     http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""
+APIClient library
+================
+
+**Module name:** :mod:`strawberryfields.api_client`
+
+.. currentmodule:: strawberryfields.api_client
+
+
+This module provides a thin client that communicates with the compute-service API over the HTTP
+protocol, based on the requests module. It also provides helper classes to facilitate interacting
+with this API via the Resource subclasses, as well as the ResourceManager wrapper around APIClient
+that is available for each resource.
+
+A single APIClient instance can be used throughout one's session in the application.
```

Comment:
Based on my understanding it looks like a default client is created in the `ResourceManager`, and then the user can configure the client using environment variables or a file. If they want more custom functionality they can create their own `APIClient` instance and pass it down to each resource. Is this the idea? It might be good to have some examples of how the user is intended to configure this in the module docs.

Reply:
I will add more information about this in the docs. In theory, this library is leaving it to the user to decide how they want to initialize the client, for whatever reasons they may have. There are safe fallback defaults if the user chooses not to do anything.